src/orders/schemas/order.schema.ts

```diff
@@ -48,3 +54,4 @@ export class Order extends Document {
 }
 
 export const OrderSchema = SchemaFactory.createForClass(Order);
+OrderSchema.index({ pickupLocation: '2dsphere' }, { unique: true });
```

Comment:
try do it in such way: 

class Location {
  @Prop({ required: false, type: Array })
  coordinates: [number, number];

  @Prop({ required: false, type: String, enum: ['Point'], default: 'Point' })
  type: string;
}

  @ApiProperty({ required: false, type: String })
  @Prop({
    required: false,
    type: Location,
  })
  loc: Location;

OrderSchema.index({ loc: '2dsphere' });